elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/verify/DynamicConfigVerifier.java

```diff
@@ -0,0 +1,157 @@
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.contrib.dynamicconfighelpers.verify;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.MissingOptionException;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.io.FileUtils;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.cert.Certificate;
+import java.util.Base64;
+
+/**
+ * Util class to Verify model tar.gz file's RSA signature with available public key in key store.
+ */
+@Slf4j
+public class DynamicConfigVerifier {
+
+    /**
+     * Main Method to Verify Signature of Model Tar file.
+     * @param args : expects 3 arguments.
+     * @throws ParseException
+     * @throws IOException
+     * @throws KeyStoreException
+     * @throws FileNotFoundException
+     * @throws SignatureException
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeyException
+     * @throws MissingOptionException
+     */
+    public static void main(String[] args) throws ParseException, InvalidKeyException, NoSuchAlgorithmException,
+            SignatureException, FileNotFoundException, KeyStoreException, IOException {
+
+        Options options = prepareOptions();
+        CommandLine cli = new DefaultParser().parse(options, args);
+
+        if (cli.hasOption("help")) {
+            printHelp(options);
+            return;
+        }
+        if (!cli.hasOption("tarFile") || !cli.hasOption("signatureFile") || !cli.hasOption("publicKeyName")) {
+            printHelp(options);
+            throw new MissingOptionException("Missing required option");
+        }
+
+        String modelTarFile = cli.getOptionValue("tarFile");
+        String signatureFile = cli.getOptionValue("signatureFile");
+        String publicKeyName = cli.getOptionValue("publicKeyName");
+
+        if (verify(readTarContents(modelTarFile), signatureFile, getPublicKey(publicKeyName))) {
+            log.info("Successfully Validated " + modelTarFile);
+        }
+        else {
+            log.error("Could not verify " + modelTarFile + " with details provided");
```

Comment:
You should set the exit status on failure to something that will cause this to abort shell scripts.